packages/scripts/src/lib/_auto-nocheck.ts

```diff
@@ -0,0 +1,160 @@
+import { relative, dirname } from 'node:path';
+import { readFile, writeFile } from 'node:fs/promises';
+import { fileURLToPath } from 'node:url';
+import resolvePkg from 'resolve';
+import globPkg from 'glob';
+import yargs from 'yargs';
+import ora from 'ora';
+import { type ProjectAnalysis } from '@glint/core';
+
+const globSync = globPkg.sync;
+const resolve = resolvePkg.sync;
+
+export async function autoNocheck(
+  args: Array<string>,
+  { cwd = process.cwd(), spinner = ora() } = {}
+): Promise<void> {
+  let glint = await loadGlintCore(cwd);
+  let { globs, explanation } = parseArgs(args);
+
+  spinner.start('Starting Glint language service...');
+
+  let project = glint.analyzeProject(cwd);
+  let fileUpdates = new Map<string, string>();
+
+  for (let filePath of collectFilePaths(globs, cwd)) {
+    spinner.text = `Checking ${relative(cwd, filePath)}...`;
+
+    let fileContents = await readFile(filePath, 'utf-8');
+    let templatesWithErrors = findTemplatesWithErrors(glint, filePath, fileContents, project);
+    if (templatesWithErrors.size) {
+      fileUpdates.set(
+        filePath,
+        insertNocheckComments(filePath, fileContents, explanation, templatesWithErrors)
+      );
+    }
+  }
+
+  project.shutdown();
+
+  spinner.text = 'Writing `@glint-nocheck` comments...';
+
+  await writeFileContents(fileUpdates);
+
+  spinner.succeed(`Done! ${fileUpdates.size} files updated.`);
+}
+
+function parseArgs(args: Array<string>): { globs: Array<string>; explanation: string } {
+  return yargs(args)
+    .scriptName('auto-nocheck')
+    .command('$0 <globs...>', 'Write a description here', (command) =>
+      command
+        .positional('globs', {
+          type: 'string',
+          array: true,
+          describe: 'One or more paths or globs specifying the files to act on.',
+          demandOption: true,
+        })
+        .option('explanation', {
+          type: 'string',
+          default: 'not typesafe yet',
+          describe: 'The explanation to be included in @glint-nocheck comments',
+        })
+    )
+    .wrap(100)
+    .strict()
+    .parseSync();
+}
+
+type GlintCore = typeof import('@glint/core') &
+  typeof import('@glint/core/lib/language-server/util/index.js');
+
+function findImport(path: string, basedir: string): string {
+  let resolvedPath = resolve(path, { basedir });
+  let directory = dirname(fileURLToPath(import.meta.url));
+  return relative(directory, resolvedPath);
+}
+
+async function loadGlintCore(cwd: string): Promise<GlintCore> {
+  try {
+    let [core, util] = await Promise.all([
+      import(findImport('@glint/core', cwd)),
+      import(findImport('@glint/core/lib/language-server/util/index.js', cwd)),
+    ]);
+
+    return { ...core, ...util };
+  } catch {
+    throw new Error(
+      `This script requires @glint/core >= 0.9.6 to be available in your project in order to run.`
+    );
```

Comment:
Does putting it in `peerDependencies` make sense, or *no* because it is apt to be run as a standalone script rather than actually installed in a project?

Reply:
I think no, for exactly that reason. I've had to open enough PRs over the years to remove codemods and the like from package dependencies that if I had my way it would be illegal to install these things as a persistent dependency in a project 😂